src/DiagnosticMessages.ts

```diff
@@ -738,85 +891,99 @@ export let DiagnosticMessages = {
+        legacyCode: 1141,
+        severity: DiagnosticSeverity.Error,
+        code: 'argument-type-mismatch'
     }),
     returnTypeMismatch: (actualTypeString: string, expectedTypeString: string, data?: TypeCompatibilityData) => ({
-        message: `Type '${actualTypeString}' is not compatible with declared return type '${expectedTypeString}'${typeCompatibilityMessage(actualTypeString, expectedTypeString, data)}'`,
+        message: `Type '${actualTypeString}' is not compatible with declared return type '${expectedTypeString}'${typeCompatibilityMessage(actualTypeString, expectedTypeString, data)} '`,
         data: data,
-        code: 1142,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1142,
+        severity: DiagnosticSeverity.Error,
+        code: 'return-type-mismatch'
     }),
     assignmentTypeMismatch: (actualTypeString: string, expectedTypeString: string, data?: TypeCompatibilityData) => ({
         message: `Type '${actualTypeString}' is not compatible with type '${expectedTypeString}'${typeCompatibilityMessage(actualTypeString, expectedTypeString, data)}`,
         data: data,
-        code: 1143,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1143,
+        severity: DiagnosticSeverity.Error,
+        code: 'assignment-type-mismatch'
     }),
     operatorTypeMismatch: (operatorString: string, firstType: string, secondType = '') => ({
         message: `Operator '${operatorString}' cannot be applied to type${secondType ? 's' : ''} '${firstType}'${secondType ? ` and '${secondType}'` : ''}`,
-        code: 1144,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1144,
+        severity: DiagnosticSeverity.Error,
+        code: 'operator-type-mismatch'
     }),
     incompatibleSymbolDefinition: (symbol: string, scopeName: string) => ({
         message: `'${symbol}' is incompatible across these scopes: ${scopeName}`,
-        code: 1145,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1145,
+        severity: DiagnosticSeverity.Error,
+        code: 'incompatible-definition'
     }),
     memberAccessibilityMismatch: (memberName: string, accessModifierFlag: SymbolTypeFlag, definingClassName: string) => ({
         message: `Member '${memberName}' is ${accessModifierNameFromFlag(accessModifierFlag)}${accessModifierAdditionalInfo(accessModifierFlag, definingClassName)}`, // TODO: Add scopes where it was defined
-        code: 1146,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1146,
+        severity: DiagnosticSeverity.Error,
+        code: 'member-accessibility-mismatch'
     }),
     typecastStatementMustBeDeclaredAtStart: () => ({
         message: `'typecast' statement must be declared at the top of the file or beginning of function or namespace`,
-        code: 1147,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1147,
+        severity: DiagnosticSeverity.Error,
+        code: 'typecast-must-be-at-start'
```

Comment:
```suggestion
        code: ''unexpected-typecast'
```

Make this more generic to "some statement must be declared at top of file or beginning of function or namespace". In case we end up with other statements like this.